### DISCUSSION TITLE
Looking for menu library SSD1309 + u8g2 over ARM Linux

### DISCUSSION THREAD

**Discussion by MikePooh** [olikraus/u8g2#2022] [General]
Hi!
I am running Orange Pi PC+ with Armbian OS. I've successfully launch SSD1309 with u8g2 Arm Linux port with C++ over harware SPI. Now I need a menu library to create user interface. There is a number of that, like [GEM](https://github.com/Spirik/GEM) or [ArduinoMenu](https://github.com/neu-rah/ArduinoMenu), however they are designed for Arduino, not ARM Linux. I was trying to rebuild them, but not sucessed since they depends on Arduino specific libraries. Can someone advice the menu library which can run with my hardware out of the box? Or give me some keys how to rebuild mentioned libraries. Thanks in advance.

**Comment by olikraus** [comment]
You could use MUI, which is part of u8g2: https://github.com/olikraus/u8g2/issues/1747
MUI is still beta, but should already work quite good.
